dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -455,37 +475,73 @@ public void close() {}
 
     private static final class ResponseInputStream extends FilterInputStream {
 
-        // Client reference is used to prevent premature termination
-        @Nullable
-        private ApacheHttpClientChannels.CloseableClient client;
-
         private final HttpClientResponse response;
 
-        ResponseInputStream(
-                @Nullable ApacheHttpClientChannels.CloseableClient client,
-                InputStream stream,
-                HttpClientResponse response) {
+        ResponseInputStream(InputStream stream, HttpClientResponse response) {
             super(stream);
-            this.client = client;
             this.response = response;
         }
 
         @Override
-        public void close() throws IOException {
-            try {
-                try {
-                    response.close();
-                } finally {
-                    super.close();
```

Comment:
This `super.close()` call is unnecessary as it may attempt to drain remaining bytes, which is already handled by response.close(). super.close may throw if response.close chose to terminate the connection rather than fully draining.